app/src/main/java/com/framgia/wsm/utils/validator/Validator.java

```diff
@@ -158,39 +159,40 @@ private boolean validate(Object factor, Rule[] rules, boolean isOptional) {
         return isValid;
     }
 
-    public Subscription initNGWordPattern() {
-        return Observable.create(new Observable.OnSubscribe<Pattern>() {
+    public Disposable initNGWordPattern() {
+        return Observable.create(new ObservableOnSubscribe<Pattern>() {
             @Override
-            public void call(Subscriber<? super Pattern> subscriber) {
-                try {
-                    BufferedReader reader = new BufferedReader(
-                            new InputStreamReader(mContext.getAssets().open("ng-word")));
-                    StringBuffer buffer = new StringBuffer();
-                    String line;
-                    while ((line = reader.readLine()) != null) {
-                        if (buffer.length() == 0) {
-                            buffer.append("(").append(line.toLowerCase(Locale.ENGLISH));
-                        } else {
-                            buffer.append("|").append(line.toLowerCase(Locale.ENGLISH));
-                        }
+            public void subscribe(@NonNull ObservableEmitter<Pattern> emitter) throws Exception {
+                BufferedReader reader = new BufferedReader(
+                        new InputStreamReader(mContext.getAssets().open("ng-word")));
+                StringBuffer buffer = new StringBuffer();
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    if (buffer.length() == 0) {
+                        buffer.append("(").append(line.toLowerCase(Locale.ENGLISH));
```

Comment:
WARNING: [AppendCharacterWithChar] Avoid appending characters as strings in StringBuffer.append.
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/strings.html#AppendCharacterWithChar